packages/network/src/logic/TrackerConnector.ts

```diff
@@ -0,0 +1,65 @@
+import { Utils } from 'streamr-client-protocol'
+import { StreamIdAndPartition, TrackerInfo } from '../identifiers'
+import { NodeToTracker } from '../protocol/NodeToTracker'
+import { Logger } from '../helpers/Logger'
+import { PeerInfo } from '../connection/PeerInfo'
+import { TrackerId } from './Tracker'
+
+export class TrackerConnector {
+
+    private readonly getStreams: () => ReadonlyArray<StreamIdAndPartition>
+    private readonly nodeToTracker: NodeToTracker
+    private readonly trackerRegistry: Utils.TrackerRegistry<TrackerInfo>
+    private readonly logger: Logger
+    private maintenanceTimer?: NodeJS.Timeout | null
+    private readonly maintenanceInterval: number
+    private connectionErrors: Map<TrackerId,boolean>
```

Comment:
Would a `Set` be more idiomatic here?  Also name could be more indicative of what the role is?
```suggestion
    private trackersWithErrorsSilenced: Set<TrackerId>
```

Reply:
Fixed